src/main/java/net/minecraftforge/event/ForgeEventFactory.java

```diff
@@ -699,4 +701,13 @@ public static boolean onPistonMovePost(World world, BlockPos pos, Direction dire
     {
         return MinecraftForge.EVENT_BUS.post(new PistonEvent.Post(world, pos, direction, extending ? PistonEvent.PistonMoveType.EXTEND : PistonEvent.PistonMoveType.RETRACT));
     }
+
+    public static void onJigsawStructurePiecesInit(JigsawStructurePoolInitEvent event)
+    {
+        MinecraftForge.EVENT_BUS.post(event);
+        for (Map.Entry<ResourceLocation, ImmutableList<ResourceLocation>> entry: event.getRemoveBuildings().entrySet())
```

Comment:
Pass the fields needed into this event so you dont have to have a public remove/add

Reply:
should this mean to add JigsawLists (JigsawPattern#field_214952_d) into the event? or just compute the removal & addition inthe event and not from the ForgeEventFactory?